atmos_model.F90

```diff
@@ -2431,932 +2447,254 @@ subroutine setup_exportdata()
+            ! MEAN NET sfc nir diffused flux (W/m**2)
+            case ('mean_net_sw_ir_dif_flx')
+              call block_data_copy(datar82d, GFS_data(nb)%coupling%nnirdf_cpl, Atm_block, nb, scale_factor=rtime, rc=localrc)
+            ! MEAN NET sfc uv+vis direct flux (W/m**2)
+            case ('mean_net_sw_vis_dir_flx')
+              call block_data_copy(datar82d, GFS_data(nb)%coupling%nvisbm_cpl, Atm_block, nb, scale_factor=rtime, rc=localrc)
+            ! MEAN NET sfc uv+vis diffused flux (W/m**2)
+            case ('mean_net_sw_vis_dif_flx')
+              call block_data_copy(datar82d, GFS_data(nb)%coupling%nvisdf_cpl, Atm_block, nb, scale_factor=rtime, rc=localrc)
+            ! MEAN precipitation rate (kg/m2/s)
+            case ('mean_prec_rate')
+              call block_data_copy(datar82d, GFS_data(nb)%coupling%rain_cpl, Atm_block, nb, scale_factor=rtimek, rc=localrc)
+            ! MEAN snow precipitation rate (kg/m2/s)
+            case ('mean_fprec_rate')
+              call block_data_copy(datar82d, GFS_data(nb)%coupling%snow_cpl, Atm_block, nb, scale_factor=rtimek, rc=localrc)
+            ! oceanfrac used by atm to calculate fluxes
+            case ('openwater_frac_in_atm')
+              call block_data_combine_fractions(datar82d, GFS_data(nb)%sfcprop%oceanfrac, GFS_Data(nb)%sfcprop%fice, Atm_block, nb, rc=localrc)
+            !--- Dycore quantities
+            ! bottom layer temperature (t)
+            case('inst_temp_height_lowest')
+              call block_data_copy_or_fill(datar82d, DYCORE_data(nb)%coupling%t_bot, zeror8, Atm_block, nb, rc=localrc)
+            ! bottom layer specific humidity (q)
+            !    !    ! CHECK if tracer 1 is for specific humidity     !    !    !
+            case('inst_spec_humid_height_lowest')
+              call block_data_copy_or_fill(datar82d, DYCORE_data(nb)%coupling%tr_bot, 1, zeror8, Atm_block, nb, rc=localrc)
+            ! bottom layer zonal wind (u)
+            case('inst_zonal_wind_height_lowest')
+              call block_data_copy_or_fill(datar82d, DYCORE_data(nb)%coupling%u_bot, zeror8, Atm_block, nb, rc=localrc)
+            ! bottom layer meridionalw wind (v)
+            case('inst_merid_wind_height_lowest')
+              call block_data_copy_or_fill(datar82d, DYCORE_data(nb)%coupling%v_bot, zeror8, Atm_block, nb, rc=localrc)
+            ! bottom layer pressure (p)
+            case('inst_pres_height_lowest')
+              call block_data_copy_or_fill(datar82d, DYCORE_data(nb)%coupling%p_bot, zeror8, Atm_block, nb, rc=localrc)
+            ! bottom layer height (z)
+            case('inst_height_lowest')
+              call block_data_copy_or_fill(datar82d, DYCORE_data(nb)%coupling%z_bot, zeror8, Atm_block, nb, rc=localrc)
+            !--- JEDI fields
+            case ('u')
+              call block_atmos_copy(datar83d, Atm(mygrid)%u, Atm_block, nb, rc=localrc)
```

Comment:
Do you run fv-gocart test with 64bit?  the Atm(mygrid)%u can be real(4) is compiled with 32BIT=Y.

Reply:
This block is not used when coupling with GOCART.

Reply:
We may have issue for this with 32BIT=Y, but this can be fixed later.